tests/test_mcp_client.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2025 The HuggingFace Team. All rights reserved.
```

Comment:
not a big fan of this test file (lots of mocks, quite verbose and hard to read). 

Would prefer to remove it and not add tests for the allowed_tools (MCP client entirely is not tested/experimental).